packages/bundler/src/bundler.ts

```diff
@@ -64,21 +68,26 @@ export class Bundler implements Subscribable<BundlerMessage, undefined> {
 
   static *create(bundles: Array<BundleOptions>): Operation<Bundler> {
     let bundler = new Bundler();
-    let rollup = watch(prepareRollupOptions(bundles));
-    let events = Subscribable
-      .from(on<Array<RollupWatcherEvent>>(rollup, 'event'))
-      .map(([event]) => event)
-      .filter(event => event.code === 'END' || event.code === 'ERROR')
-      .map(event => event.code === 'ERROR' ? { type: 'error', error: event.error } : { type: 'update' });
 
     return yield resource(bundler, function*() {
-      try {
-        yield events.forEach(function*(message) {
-          bundler.channel.send(message as BundlerMessage);
-        });
-      } finally {
-        rollup.close();
-      }
+      let rollup: RollupWatcher | null = null;
+
+      yield ensure(() => {
+        if (rollup) {
+          rollup.close();
+        }
+      });
+
+      rollup = watch(prepareRollupOptions(bundles));
+      let events: ChainableSubscription<Array<RollupWatcherEvent>, undefined> = yield subscribe(on<Array<RollupWatcherEvent>>(rollup, 'event'));
```

Comment:
I'm still not quite sure why a try/catch won't catch the right thing. If you put it around 82-90, it will catch any exception raised therein.

Reply:
This is what it was originally:

```ts
let rollup: RollupWatcher = watch(prepareRollupOptions(bundles));;
let events: ChainableSubscription<Array<RollupWatcherEvent>, undefined> = yield subscribe(on<Array<RollupWatcherEvent>>(rollup, 'event'));
let messages = events
  .map(([event]) => event)
  .filter(event => event.code === 'END' || event.code === 'ERROR')
  .map(event => event.code === 'ERROR' ? { type: 'error', error: event.error } : { type: 'update' });

try {
  yield messages.forEach(function*(message) {
    bundler.channel.send(message as BundlerMessage);
  });
} finally {
  rollup.close();
}
```

Which is wrong I think, prompting me to make a change. However, rather than just wrap more in the `try` block I thought `ensure()` was the way to go, but now that you've explained that that's not the way to go I'll update to this:

```ts
let rollup: RollupWatcher = watch(prepareRollupOptions(bundles));;

try {
  let events: ChainableSubscription<Array<RollupWatcherEvent>, undefined> = yield subscribe(on<Array<RollupWatcherEvent>>(rollup, 'event'));
  let messages = events
    .map(([event]) => event)
    .filter(event => event.code === 'END' || event.code === 'ERROR')
    .map(event => event.code === 'ERROR' ? { type: 'error', error: event.error } : { type: 'update' });

  yield messages.forEach(function*(message) {
    bundler.channel.send(message as BundlerMessage);
  });
} finally {
  rollup.close();
}
```